### DISCUSSION TITLE
Sync multiple RTSP link by mediamtx

### DISCUSSION THREAD

**Discussion by brusezot** [bluenviron/mediamtx#3403] [Questions]
### Question

I've developed an app that lets you add 9 video files that ffmpeg plays back in a loop and that mediamtx retrieves as an RTSP link, but how can I ensure that the streams are perfectly synced down to the frame? Thanks for your feedback. It's for a lab

**Comment by aler9** [comment]
Hello, i've dealt with a similar scenario, first of all you have to make sure that all the video files have the same FPS and duration, with a millisecond precision at least. If you are unsure, truncate them:

```
ffmpeg -i file.mp4 -t 00:40:05.555 -c copy file2.mp4
```

Where `00:40:05.555` is the minimum duration of any file.

This allows to run videos in sync for about 24 hours. After this, you have to restart all FFmpeg instances at once. I'd like to have a cleaner solution but the issue is completely on the client side, we can't do anything since we control the server side.

[a comment by an automated account is not shown]
